src/io/mwaf.rs

```diff
@@ -5,94 +5,80 @@
     /// The number of correlator fine channels per flag file, and the `NCHANS` key from the primary hdu.
-    pub num_channels: usize,
+    pub num_channels: u32,
     /// Total number of antennas (tiles) in the array, and the `NANTENNA` key from the primary hdu
-    pub num_ants: usize,
+    pub num_ants: u32,
     /// Number of timesteps in the observation, and the `NSCANS` key from the primary hdu
-    pub num_timesteps: usize,
+    pub num_timesteps: u32,
     /// The `NPOLS` key from the primary hdu
-    pub num_pols: usize,
-    /// The `GPUBOXNO` key from the primary hdu
-    pub gpubox_id: usize,
-    /// The `COTVER` key from the primary hdu
-    pub cotter_version: String,
-    /// The `COTVDATE` key from the primary hdu
-    pub cotter_version_date: String,
+    pub num_pols: u8,
+    /// The name of the software used to generate this flag file.
+    pub software: String,
     /// The width of each fine channel mask vector in bytes, or the `NAXIS1` key from the table hdu
-    pub bytes_per_row: usize,
+    pub bytes_per_row: u32,
     /// The number of rows (timesteps × baselines), and the `NAXIS2` key from the table hdu.
-    pub num_rows: usize,
-    // TODO: is it useful to output aoflagger version and strategy?
-}
-
-impl FlagFileHeaders {
-    /// Construct the [`FlagFileHeaders`] struct corresponding to the provided mwalib context and
-    /// gpubox id.
-    pub fn from_gpubox_context(gpubox_id: usize, context: &CorrelatorContext) -> Self {
-        let num_fine_per_coarse = context.metafits_context.num_corr_fine_chans_per_coarse;
-        Self {
-            version: "1.0".to_string(),
-            obs_id: context.metafits_context.obs_id,
-            num_channels: context.metafits_context.num_corr_fine_chans_per_coarse,
-            num_ants: context.metafits_context.num_ants,
-            num_timesteps: context.num_common_timesteps,
-            num_pols: 1,
-            gpubox_id,
-            cotter_version: format!("Birli-{}", crate_version!()),
-            // TODO: use something like https://github.com/rustyhorde/vergen
-            cotter_version_date: "2021-04-14".to_string(),
-            bytes_per_row: num_fine_per_coarse / 8 + usize::from(num_fine_per_coarse % 8 != 0),
-            num_rows: context.num_common_timesteps * context.metafits_context.num_baselines,
-        }
-    }
+    pub num_rows: u32,
+    /// The version of aoflagger used to generate these flags.
+    pub aoflagger_version: Option<String>,
+    /// The aoflagger strategy used to generate these flags.
+    pub aoflagger_strategy: Option<String>,
 }
 
 /// A group of .mwaf Files for the same observation
 pub struct FlagFileSet {
-    gpubox_fptrs: BTreeMap<usize, FitsFile>,
+    /// A map associating gpubox numbers with filenames as well as the flag
+    /// counts for that gpubox (used to determine occupancy which is written to
+    /// the mwaf files after all flags are written).
+    pub(crate) gpuboxes: Vec<(usize, PathBuf, Vec<u64>)>,
```

Comment:
I would have split this into 2 separate maps, one for filenames and one for occupancy, but if it works, it works.

Reply:
I couldn't use a BTreeMap in parallel for some reason, but even if I could, having two maps seems to add complexity for no reason?  Honestly, the main benefit of a map/set structure is to have faster lookups, so here we don't really gain anything as the length is unlikely to be more than 24, therefore I think the choice of Vec is fine.